pkg/fftypes/data.go

```diff
@@ -52,6 +52,38 @@ type Data struct {
 	ValueSize int64 `json:"-"` // Used internally for message size calcuation, without full payload retrieval
 }
 
+func (br *BlobRef) BatchBlobRef(broadcast bool) *BlobRef {
+	if br == nil {
+		return nil
+	}
+	// For broadcast data the blob reference contains the "public" (shared storage) reference, which
+	// must have been allocated to this data item before sealing the batch.
```

Comment:
This is the operative decision for #568 - I'm still wondering if it's possible for the public reference to be removed from the batch hash altogether. Perhaps it could be store on the `Blob` instead of on the `BlobRef`?

Reply:
It's mainly odd that private blob transfer happens _after_ batch sealing, but public blob transfer must happen _before_ batch sealing in order to include the IPFS ref. Seems like they should happen in the same order regardless.

When a node receives an IPFS ref, it does do some checking of the fetched contents to verify they match the blob hash before recording the blob as received. The question is whether this is "good enough" to say we can send IPFS refs without including them in the batch's hash proof.

Reply:
> Seems like they should happen in the same order regardless.

I do not think this is possible, as IPFS does not have a "messaging" capability. It's just a storage system.